dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -380,10 +452,7 @@ public boolean isAncestorOf(UfsUrl ufsUrl) throws InvalidPathException {
     if (!Objects.equals(getScheme(), ufsUrl.getScheme())) {
       return false;
     }
-    // TODO(Tony Sun): optimize the performance later
-    // Both of the ufsUrls has the same scheme and authority, so just need to compare their paths.
-    return PathUtils.hasPrefix(PathUtils.normalizePath(ufsUrl.getFullPath(), SLASH_SEPARATOR),
-        PathUtils.normalizePath(getFullPath(), SLASH_SEPARATOR));
+    return PathUtils.hasPrefix(ufsUrl.getFullPath(), this.getFullPath());
```

Comment:
Why create strings out of path component lists and then check for string prefix, while you already have the path components list? just compare the lists.

Reply:
This is an old flaw I forget to remove. In an Old design, the path components list contains empty elements representing "/", and I do not remove redundant slashes like "/tmp///a", so there are many empty elements, while it equals "/tmp/a". I will fix it.

Reply:
Wow, how stupid the code. I'll fix it.